azurerm/internal/services/compute/tests/resource_arm_disk_encryption_set_test.go

```diff
@@ -0,0 +1,243 @@
+package tests
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2018-02-14/keyvault"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMDiskEncryptionSet_basic(t *testing.T) {
+	resourceName := "azurerm_disk_encryption_set.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(6)
+	resourceGroup := fmt.Sprintf("acctestRG-%d", ri)
+	vaultName := fmt.Sprintf("vault%d", ri)
+	keyName := fmt.Sprintf("key-%s", rs)
+	desName := fmt.Sprintf("acctestdes-%d", ri)
+	location := acceptance.Location()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDiskEncryptionSetDestroy,
+		Steps: []resource.TestStep{
+			// This test step is temporary due to freezing of functions in keyVault.
+			// After applying soft-delete and purge-protection in keyVault, this extra step can be removed.
+			{
+				Config:  testAccPrepareKeyvaultAndKey(resourceGroup, location, vaultName, keyName),
+				Destroy: false,
+				Check:   resource.ComposeTestCheckFunc(),
+			},
+			{
+				PreConfig: func() { enableSoftDeleteAndPurgeProtectionForKeyvault(resourceGroup, vaultName) },
+				Config:    testAccAzureRMDiskEncryptionSet_basic(resourceGroup, location, vaultName, keyName, desName),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDiskEncryptionSetExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "active_key.0.source_vault_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "active_key.0.key_url"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
```

Comment:
(as above) do we need a couple of steps here to first delete the DES, then purge/disable soft delete to allow the key vault to be torn down?